### ISSUE TITLE
Some questions in the process of implementing your code 

### ISSUE THREAD

**Issue by Clara-mm** [ml-jku/lsc#5]
Hi @AndreasMayr,

I feel sorry for disturbing you and interrupting your routine.

From URL (http://www.bioinf.jku.at/research/lsc/), I got to know that some fingerprints (such as MACCS, PubChem, and SHED )were selected to produce features.  There are many others fingerprints.  Why did you choose those fingerprints?  What's so special about those fingerprints? 

Why did you convert data to binary format?  (https://github.com/ml-jku/lsc/blob/master/commdands.sh#L66)  Wouldn't it be more convenient to read data in CSV format directly? 

As you told that 3 different models were trained. So, What is your training data? And where is it? Which part of code are training code? (Are lines 124-132 training code in commands.sh?  )

There are many assay measurements ( such as Ki, IC50, and EC50) in file "ChEMBL20-MySQL". So, I can not  figure out what's your training lables?

I would be very grateful if you could reply me.

**Comment by AndreasMayr** [comment]
> From URL (http://www.bioinf.jku.at/research/lsc/), I got to know that some fingerprints (such as MACCS, PubChem, and SHED )were selected to produce features. There are many others fingerprints. Why did you choose those fingerprints? What's so special about those fingerprints?

In general our strategy was to include commonly used fingerprint features (e.g. MACCS, ECFP is quite well known). But you may be right, that the decision which fingerprint features to use and how to merge them together may have been arbitrary to a certain degree. 

> Why did you convert data to binary format? (https://github.com/ml-jku/lsc/blob/master/commdands.sh#L66) Wouldn't it be more convenient to read data in CSV format directly?

For the SVM, which is written in C++ code, binary format was more convenient.
In general, I think, it's also faster than reading in a csv file each time.
Moreover, for sparse features, like ECFP, csv files would get extremly large, whereas
this is not an issue for the binary sparse format.

> As you told that 3 different models were trained. So, What is your training data? And where is it? Which part of code are training code? (Are lines 124-132 training code in commands.sh? )

Exactly, these are the lines where nested cluster cross validation is applied
(at least for the DNN codes).

> There are many assay measurements ( such as Ki, IC50, and EC50) in file "ChEMBL20-MySQL". So, I can not figure out what's your training lables?

We didn't use everything.

The SQL query for extracting activities is given here:
https://github.com/ml-jku/lsc/blob/master/extractChEMBL.sql

Then, I filtered out a lot of bioactivities. The script, where this is done, is given here:
https://github.com/ml-jku/lsc/blob/master/createTrainTable.R
There, we tried to define some critera, what to consider as active/inctive; then
we tried to check for contradictions, in order not to use these as training/evaluation data, etc.
The process is also described in the Supplementary Material of the publication.

**Comment by Clara-mm** [comment]
> We didn't use everything.

Do you mean to say that there are no training label in your work?  According to the Supplementary Material of the publication, binary labels (active/inactive) are assigned to compound-measurement pairs. So, The binary labels (active/inactive) are  training labels, aren't they?

> The SQL query for extracting activities is given here
> https://github.com/ml-jku/lsc/blob/master/extractChEMBL.sql

Whehter the purpose of the SQL query for extracting activities extracted from ChEMBL is to generate a benchmark datasets?

**Comment by Clara-mm** [comment]
We didn't use everything.

Do you mean to say that there are no training label in your work? According to the Supplementary Material of the publication, binary labels (active/inactive) are assigned to compound-measurement pairs. So, The binary labels (active/inactive) are training labels, aren't they?

The SQL query for extracting activities is given here
https://github.com/ml-jku/lsc/blob/master/extractChEMBL.sql

Whehter the purpose of the SQL query for extracting activities extracted from ChEMBL is to generate a benchmark datasets?

On 10/1/2020 06:42，Andreas Mayr<notifications@github.com> wrote：

From URL (http://www.bioinf.jku.at/research/lsc/), I got to know that some fingerprints (such as MACCS, PubChem, and SHED )were selected to produce features. There are many others fingerprints. Why did you choose those fingerprints? What's so special about those fingerprints?

In general our strategy was to include commonly used fingerprint features (e.g. MACCS, ECFP is quite well known). But you may be right, that the decision which fingerprint features to use and how to merge them together may have been arbitrary to a certain degree.

Why did you convert data to binary format? (https://github.com/ml-jku/lsc/blob/master/commdands.sh#L66) Wouldn't it be more convenient to read data in CSV format directly?

For the SVM, which is written in C++ code, binary format was more convenient.
In general, I think, it's also faster than reading in a csv file each time.
Moreover, for sparse features, like ECFP, csv files would get extremly large, whereas
this is not an issue for the binary sparse format.

As you told that 3 different models were trained. So, What is your training data? And where is it? Which part of code are training code? (Are lines 124-132 training code in commands.sh? )

Exactly, these are the lines where nested cluster cross validation is applied
(at least for the DNN codes).

There are many assay measurements ( such as Ki, IC50, and EC50) in file "ChEMBL20-MySQL". So, I can not figure out what's your training lables?

We didn't use everything.

The SQL query for extracting activities is given here:
https://github.com/ml-jku/lsc/blob/master/extractChEMBL.sql

Then, I filtered out a lot of bioactivities. The script, where this is done, is given here:
https://github.com/ml-jku/lsc/blob/master/createTrainTable.R
There, we tried to define some critera, what to consider as active/inctive; then
we tried to check for contradictions, in order not to use these as training/evaluation data, etc.
The process is also described in the Supplementary Material of the publication.

**Comment by AndreasMayr** [comment]
> Do you mean to say that there are no training label in your work? According to the Supplementary Material of the publication, binary labels (active/inactive) are assigned to compound-measurement pairs. So, The binary labels (active/inactive) are training labels, aren't they?

No. Of course we used training labels. Otherwise it wouldn't have made sense. But as described in the Supplementary Material 
Section 2.1. we used several filters on the raw bioactivities from ChEMBL. ChEMBL contained lots of bioactivities, but we didn't make use from all of them, because e.g. we considered some as contradicting. For some we relied on activity comments, for some there might be no comments given, etc. We also only considered certain types of "activity comments" as we didn't look manually through all.

> Whether the purpose of the SQL query for extracting activities extracted from ChEMBL is to generate a benchmark datasets?

Yes, the purpose was to generate the benchmark dataset. After retrieving the result, we used it as an input to https://github.com/ml-jku/lsc/blob/master/createTrainTable.R, which more or less implements what we described in Supplementary Material 
Section 2.1.
